code/datums/elements/drag_pickup.dm

```diff
@@ -0,0 +1,25 @@
+/datum/element/drag_pickup
+	element_flags = ELEMENT_DETACH
+
+/datum/element/drag_pickup/Attach(datum/target)
+	if(isatom(target))
+		if(ismovable(target))
+			RegisterSignal(target, COMSIG_MOUSEDROP_ONTO, .proc/pick_up)
+	return ..()
+
+/datum/element/drag_pickup/Detach(datum/source, force)
+	UnregisterSignal(source, COMSIG_MOUSEDROP_ONTO)
+	return ..()
+
+/datum/element/drag_pickup/proc/pick_up(atom/source, atom/over, mob/user)
+	SIGNAL_HANDLER
+	var/mob/living/M = user
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names